[Feature guide] Add end user's IdP information to host vitals (#26989)
fleetdm/fleet: articles/add-user-info-from-idp-to-host.md

> # Add user information from identity provider (IdP) to host

Reviewer: ```suggestion
# Foreign host vitals: identity provider (IdP) groups
```

Thoughts on this title? Mike and I have been using "Foreign vitals" a lot. I think it's a good product marketing approach. We might add more foreign vitals later

Reviewer: If that new title looks good, I think let's change the file name too so that it's used as the website URL. The website uses the filename to generate the guide's URL.

Author: @noahtalerman I changed `Foreign host vitals: identity provider (IdP) groups` to `Foreign host vitals: identity provider (IdP) user information`, since we add more than groups. Wdyt?

Reviewer: @marko-lisica I think let's go with something like "Foreign vitals: map users from your IdP to hosts"

We have "foreign vitals", mention IdP (I think we keep it as an acronym to keep the title short, and we have mapping users to hosts.

![Import users from IdP to Fleet](../website/assets/images/articles/add-users-from-idp-cover-img.png)

_Available in Fleet Premium._

> Learn how to connect your identity provider (IdP) to retrieve end-user information and map it to a
> host. Fleet uses [SCIM](https://scim.cloud/) protocol.
> 
> Below you can learn how to connect [Okta](https://www.okta.com/), [Microsoft Entra ID](https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-id) or [Authentik](https://goauthentik.io/).

Reviewer: ```suggestion
Fleet supports [Okta](#okta), [authentik](#authentik), [Microsoft Active Directory (AD) / Entra ID](#microsoft-entra-id), as well as any other identity provider (IdP) that supports the [SCIM](https://scim.cloud/) standard.

To add IdP host vitals, follow steps for your IdP.
```

I think we can make this language consistent with what we have here in the [SSO docs](https://fleetdm.com/docs/deploy/single-sign-on-sso#single-sign-on-sso):

![Screenshot 2025-03-21 at 9 35 56 AM](https://github.com/user-attachments/assets/3b3526a5-ecf0-4217-9603-18adcf076bbb)

> ## Connect Okta

Reviewer: ```suggestion
## Okta
```

2. On the same page, select **Rules** tab. Create rule that will assign users from groups that you want to provision to Fleet to newly created "Fleet human-device mapping" group.
![Okta group rule](../website/assets/images/articles/okta-scim-group-rules.png)
3. Select **Applications > Applications** in the main menu, select app that you created previously, then select **Assignements** tab.
4. Select **Assign > Assign to Groups**, then click **Assign** next to the "Fleet human-device mapping" group, then select **Done**. Now all users that you assigned to "Fleet human-device mapping" group via rule will be provisioned to Fleet. It may take a while if you have many users.
5. On the same page, select **Push Groups** tab, then select **Push Groups > Find groups by name**,
   and add all groups that you assigned via rule to "Fleet human-device mapping" group previously (make sure that
   **Push group memberships immediately** is selected). All groups will be provisioned to
   Fleet, and Fleet will map those groups to already provisioned users .

> ## Connect Microsoft Entra ID

Reviewer: ```suggestion
## Microsoft Entra ID
```

![Entra SCIM attributes mapping for groups](../website/assets/images/articles/entra-group-scim-attributes.png)    
12. Select **Provision Microsoft Entra ID Users**.
13. Ensure that the attributes `userName`, `givenName`, `familyName`, `active`, and `externalId` are
    mapped to **Microsoft Entra ID Attribute**. Currently, Fleet support only these attributes and
    they are required as well. Delete the rest of the attributes, select **Save**, and after it's
    saved, select close icon on the top right corner.
![Entra SCIM attributes mapping for users](../website/assets/images/articles/entra-user-scim-attributes.png)  
14. Select **Users and groups** from the side menu, select **+ Add user/group**, then select
    **None Selected** link.
    
15. Select users and groups that you want to add to Fleet, then select
    **Assign**. 
16. Select **Overview** from the side menu and select **Start provisioning**.

> After you started provisioning Microsoft Entra ID sends data to Fleet immediately. Subsequent
> requests are triggered every 20-40 minutes. If you edit user information in Entra, it might
> take up to 40 minutes to see the change in Fleet.

> ## Connect Google Workspace (with authentik)

Reviewer: ```suggestion
## Google Workspace
```

> Google Workspace doesn't support SCIM provisioning, so we suggest using [authentik](https://goauthentik.io/) to import users from Google Workspace and then add users to Fleet via SCIM protocol.

Reviewer: ```suggestion
Google Workspace doesn't natively support the [SCIM](https://scim.cloud/) standard. The best practice is to import users to [authentik](https://goauthentik.io/). Authentik then adds users to Fleet.
```

Reviewer: "Best practice" instead of "suggest" to help folks feel confident.

- **Expression**:
    ```
    return {
        "username": ldap.get("mail"),
    }
    ```

- **Name**: Google LDAP mail > email  
- **Expression**:
    ```
    return {
        "email": ldap.get("mail"),
    }
    ```

- **Name**: Google LDAP givenName > first_name
- **Expression**:
    ```
    return {
        "attributes": {
            "first_name": ldap.get("givenName"),
        },
    }
    ```

- **Name**: Google LDAP sn > last_name
- **Expression**:
    ```
    return {
        "attributes": {
            "last_name": ldap.get("sn"),
        },
    }
    ```

- **Name**: Google LDAP displayName > name
- **Expression**:
    ```
    return {
        "name": ldap.get("displayName")
    }
    ```

- **Name**: Google LDAP displayName > name
- **Expression**:
    ```
    return {
        "name": ldap.get("displayName")
    }
    ```

- **Name**: Google LDAP displayName to name (group)
- **Expression**:
    ```
    return {
        "name": ldap.get("displayName")
    }
    ```

- **Name**: Google LDAP objectSid to ldap_uniq (group)
- **Expression**:
    ```
    return {
        "attributes": {
            "ldap_uniq": ldap.get("objectSid"),
        },
    }

>     ```

Reviewer: @marko-lisica looks like you might be missing a closing "```" here. The markdown formatting looks off.

Author: @noahtalerman Seems as GitHub bug. I tested and it works well when I run website.

<img width="846" alt="Screenshot 2025-03-21 at 18 19 54" src="https://github.com/user-attachments/assets/2a1ee95f-1668-4c03-944a-719d3c7a7952" />
<img width="834" alt="Screenshot 2025-03-21 at 18 20 01" src="https://github.com/user-attachments/assets/f0d8a3b5-9bfc-46bf-9bdd-d71d25716a4a" />

#### 4. Configure LDAP connection in authentik

1. Select **Directory > Federation and Social login** from the side menu.
2. Select **Create**, then select **LDAP Source**, and select **Next**.
3. Add friendly name (e.g. Google LDAP)
4. Make sure that **Enable**, **Sync users** and **Sync groups** are toggled on.

> 5. In **Server URL** enter `ldap://ldap.google.com`.

Author: ```suggestion
5. In **Server URL** enter `ldap://ldap.google.com`. For more information refer to [Google docs](https://support.google.com/a/answer/9089736?hl=en&ref_topic=9173976&sjid=5482490660946222035-EU#basic-instructions).
```

# Add user information from identity provider (IdP) to host

_Available in Fleet Premium._

> Learn how to connect your identity provider (IdP) to retrieve end-user information and map it to a
> host. This simplifies the process of identifying which employee is assigned to each host.
> Fleet uses System for Cross-domain Identity Management (SCIM) protocol to retreive users from IdP. This feature is available in Fleet [v4.67.0](https://github.com/fleetdm/fleet/releases/tag/fleet-v4.67.0).

Reviewer: ```suggestion
Learn how to connect your identity provider (IdP) to retrieve end-user information and map it to a
host. Fleet uses the System for Cross-domain Identity Management (SCIM)
```

Reviewer: Add link to SCIM

> Currently, Fleet supports only Okta.

Reviewer: ```suggestion
```

Reviewer: If someone wants to try Entra then they can we just won't document it at first.
